Yosemite/Yosemite/Stores/Order/OrdersUpsertUseCase.swift

```diff
@@ -304,25 +304,25 @@ struct OrdersUpsertUseCase {
     /// Updates, inserts, or prunes the provided `storageOrder`'s custom fields using the provided `readOnlyOrder`'s custom fields
     ///
     private func handleOrderCustomFields(_ readOnlyOrder: Networking.Order, _ storageOrder: Storage.Order, _ storage: StorageType) {
-        // Upsert the `customFields` from the `readOnlyOrder`
-        readOnlyOrder.customFields.forEach { readOnlyCustomField in
-            if let existingStorageMetaData = storage.loadOrderMetaData(siteID: readOnlyOrder.siteID,
-                                                                       orderID: storageOrder.orderID,
-                                                                       metadataID: readOnlyCustomField.metadataID) {
-                existingStorageMetaData.update(with: readOnlyCustomField)
-            } else {
-                let newStorageMetaData = storage.insertNewObject(ofType: Storage.MetaData.self)
-                newStorageMetaData.update(with: readOnlyCustomField)
-                storageOrder.addToCustomFields(newStorageMetaData)
-            }
-        }
+        // Create a set of metadata IDs from the read-only order for quick lookup
+        let readOnlyMetadataIDs = Set(readOnlyOrder.customFields.map { $0.metadataID })
 
-        // Now, remove any objects that exist in `storageOrder.customFields` but not in `readOnlyOrder.customFields`
+        // Remove any objects that exist in `storageOrder.customFields` regarding a specific order.
         storageOrder.customFields?.forEach { storageCustomField in
-            if readOnlyOrder.customFields.first(where: { $0.metadataID == storageCustomField.metadataID } ) == nil {
-                storageOrder.removeFromCustomFields(storageCustomField)
-                storage.deleteObject(storageCustomField)
-            }
+            storageOrder.removeFromCustomFields(storageCustomField)
+            storage.deleteObject(storageCustomField)
+        }
+
+        // Create `customFields` objects from the `readOnlyOrder`
```

Comment:
I think we should unit tests to `OrdersUpsertUseCaseTests` to validate that it can handle a large number of `MetaData` items. Right now, I see only tests that add/remove 1 custom field. What do you think?

Reply:
Implemented here dd9ca2356ceb1927721114c57c770a18409939ed